src/fit.jl

```diff
@@ -92,15 +92,25 @@ function train!(
     start_iter = get_iter_number(estimator) + 1
     end_iter = start_iter + num_iterations - 1
 
-    for iter in start_iter:end_iter
+    metrics_idx_sequence = (((start_iter:end_iter) .- 1) .% estimator.metric_freq) .== 0
+    total_metrics_evals = sum(metrics_idx_sequence)
+    metric_idx = 0
+
+    for (idx, iter) in enumerate(start_iter:end_iter)
+
         is_finished = LGBM_BoosterUpdateOneIter(estimator.booster)
-        log_debug(verbosity, Dates.CompoundPeriod(now() - start_time),
-                  " elapsed, finished iteration ", iter, "\n")
+
+        log_debug(verbosity, Dates.CompoundPeriod(now() - start_time), " elapsed, finished iteration ", iter, "\n")
+
+        metric_idx = metric_idx + metrics_idx_sequence[idx]
```

Comment:
`+=` ?

Reply:
I don't personally prefer this way of doing it, so I'll leave it In this case I think